### ISSUE TITLE
Sub dependency build error

### ISSUE THREAD

**Issue by idokleinman** [Carthage/Carthage#1173]
I have a carthage lib (spark-sdk-ios) which depends on AFNetworking framework.
So I try to integrate it in a new project, my Cartfile is:
`github "spark/spark-sdk-ios" "master"`

and the output for `carthage update --platform iOS --use-submodules --no-use-binaries` is:

```
The following build commands failed:
    CompileC /Users/ido/Library/Developer/Xcode/DerivedData/spark-sdk-dmjaxlhdvmmndrddoihgcyywmqgg/Build/Intermediates/ParticleSDK.build/Release-iphoneos/ParticleSDK.build/Objects-normal/armv7/SparkDevice.o /Users/ido/projects/Spark/particle-sdk-ios/test_cart/Carthage/Checkouts/spark-sdk-ios/Pod/Classes/SDK/SparkDevice.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
/Users/ido/Library/Developer/Xcode/DerivedData/spark-sdk-dmjaxlhdvmmndrddoihgcyywmqgg/Build/Products/Release-iphoneos/AFNetworking.framework/Headers/AFNetworking.h:50:9: fatal error: 'AFNetworking/AFAutoPurgingImageCache.h' file not found
/Users/ido/projects/Spark/particle-sdk-ios/test_cart/Carthage/Checkouts/spark-sdk-ios/Pod/Classes/SDK/SparkDevice.m:11:9: fatal error: could not build module 'AFNetworking'
A shell task failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
    CompileC /Users/ido/Library/Developer/Xcode/DerivedData/spark-sdk-dmjaxlhdvmmndrddoihgcyywmqgg/Build/Intermediates/ParticleSDK.build/Release-iphoneos/ParticleSDK.build/Objects-normal/armv7/SparkDevice.o /Users/ido/projects/Spark/particle-sdk-ios/test_cart/Carthage/Checkouts/spark-sdk-ios/Pod/Classes/SDK/SparkDevice.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

AFNetworking by itself builds just fine.. 
What am I doing wrong?

**Comment by mdiep** [comment]
This appears to be the same issue as #1161. Carthage is trying to build the shared dynamic framework for `ParticleSDK` in `Example/spark-sdk.xcworkspace`, but that scheme doesn't build out of the box—it requires CocoaPods.

```
/usr/bin/xcrun xcodebuild -workspace Carthage/Checkouts/spark-sdk-ios/Example/spark-sdk.xcworkspace -scheme ParticleSDK -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build
…
** BUILD FAILED **
```

**Comment by idokleinman** [comment]
Revisiting this issue - actually the `ParticleSDK` target has AFNetworking sub-project (as framework) so should not depend on Cocoapods at all. I still don't understand why the build error. 

Another input: weird behavior back when origin/master was pointing at my "0.4.0" label - `Cartfile` containing `"spark-sdk-ios" "0.4.0"` was failing and `"spark-sdk-ios" "master"` was building fine - same commit. Same target. Any ideas?

**Comment by mdiep** [comment]
> Another input: weird behavior back when origin/master was pointing at my "0.4.0" label - Cartfile containing "spark-sdk-ios" "0.4.0" was failing and "spark-sdk-ios" "master" was building fine - same commit. Same target. Any ideas?

I think we sometimes fail to fetch `master`. You may need to use `carthage fetch`.

**Comment by idokleinman** [comment]
1. Should I add something to the Cartfile in regards to the `fetch`? why?
2. the main issue - how do I go about publishing a carthage lib that depends on another carthage framework? (such as in my case my lib builds on AFNetworking) - the main shared scheme builds a target that has `AFNetworking.framework` as linked framework - builds well in XCode and fails when Carthage is checking out and builds it. Am I doing anything out of the ordinary here?

**Comment by mdiep** [comment]
> Should I add something to the Cartfile in regards to the fetch? why?

No. There's just a bug where Carthage sometimes fails to fetch if your `Cartfile` specifies a branch instead of a SHA, tag, or release.

> the main issue - how do I go about publishing a carthage lib that depends on another carthage framework? (such as in my case my lib builds on AFNetworking) - the main shared scheme builds a target that has AFNetworking.framework as linked framework - builds well in XCode and fails when Carthage is checking out and builds it. Am I doing anything out of the ordinary here?

It looks like `AFNetworking` is building correctly on `master` now. But now it fails because your shared `Spark-SDK` scheme is "missing"—if you open `spark-sdk.xcworkspace`, you can't build that scheme. This is probably because it's generated by CocoaPods.

To work with Carthage you need to have a dynamic framework scheme that's shared from an Xcode project or workspace and builds without running any CocoaPods commands.

**Comment by idokleinman** [comment]
but that's the thing - in that `.xcworkspace` I have only 1 shared scheme called "ParticleSDK" with a target which is a dynamic framework that has no Cocoapod dependencies - I added `AFNetworking` for this target as an embedded framework, there's no Cocoapod commands for this target at all. I also set a "test_cart1" tag to the tip of my test branch and modified the `Cartfile` accordingly to rule out the option we are failing on the bug you mentioned. Truly can't understand why build is failing.
